app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperFileManager.kt

```diff
@@ -31,19 +31,21 @@ class WallpaperFileManager(
      * files for each of a portrait and landscape orientation as well as a thumbnail.
      */
     suspend fun lookupExpiredWallpaper(name: String): Wallpaper? = withContext(Dispatchers.IO) {
-        if (getAllLocalWallpaperPaths(name).all { File(storageRootDirectory, it).exists() }) {
+        if (allAssetsExist(name)) {
             Wallpaper(
                 name = name,
                 collection = Wallpaper.DefaultCollection,
                 textColor = null,
                 cardColor = null,
+                thumbnailFileState = Wallpaper.ImageFileState.Downloaded,
+                assetsFileState = Wallpaper.ImageFileState.Downloaded,
             )
         } else null
     }
 
-    private fun getAllLocalWallpaperPaths(name: String): List<String> =
-        Wallpaper.ImageType.values().map { orientation ->
-            getLocalPath(name, orientation)
+    private fun allAssetsExist(name: String): Boolean =
+        Wallpaper.ImageType.values().fold(true) { acc, type ->
+            acc && File(storageRootDirectory, getLocalPath(name, type)).exists()
         }
```

Comment:
I haven't seen `fold` used before. What's the TLDR of this block?

Reply:
Check if each file exists and accumulate those results into a single value

Reply:
ahh gotcha gotcha. `Wallpaper.ImageType.values().all { }` would have also worked in this situation

Reply:
Oh sure, that's actually much more clear too. I'll update accordingly.